Use only Executor in ThreadedActionListener

danielmitterdorfer opened the pull request:
ThreadedActionListener only relies on methods provided by `Executor` but required to pass an `ExecutorService`. With this commit we relax the required type to what's actually needed.

Relates #93184

danielmitterdorfer committed:
Use only Executor in ThreadedActionListener

ThreadedActionListener only relies on methods provided by `Executor` but
required to pass an `ExecutorService`. With this commit we tighten the
required type to what's actually needed.

DaveCTurner reviewed (approved):
LGTM

DaveCTurner commented:
Not sure about backporting this to 8.7 tho, that seems unnecessary (unless there's a bug fix which needs it?)

Also wording nit:

> With this commit we tighten the required type to what's actually needed.

We're _relaxing_ the required type :)

danielmitterdorfer commented:
Thanks for the quick review! 

> Not sure about backporting this to 8.7 tho, that seems unnecessary (unless there's a bug fix which needs it?)

No, it's not needed from my perspective and I thought there's no harm. My intention was to avoid surprises in any future backports to 8.7. Anyway, I've removed the respective labels now and merge the PR only to `main`. 

> We're relaxing the required type :)

Gah. Corrected as well... :)
